### ISSUE TITLE
_include with modifier is not recognized as an include by Add

### ISSUE THREAD

**Issue by ewoutkramer** [FirelyTeam/fhir-net-api#222]
Wondering how to best to handle _include:recurse= . The SearchParams.add put them as a regular parameter instead of part of _include collection. Wondering if i should add a IncludeRecurse collection.

**Comment by brianpos** [comment]
Can you add an example to include in the unit test (and i'm not quite with you)

**Comment by ewoutkramer** [comment]
Got a bug report by mail with this text, so I pasted it here as a reminder to double check this....

**Comment by gautebrakstad** [comment]
Its originally from me. Problem was that SearchParams.Include collection does not support includes with recurse modifier. Example [base]/Observation?_include:recurse=Observation.related-target
This instead get added to the regular searchparam.

**Comment by brianpos** [comment]
I agree to add another internal collection for the reverseIncludes (doing it in my server)

**Comment by ewoutkramer** [comment]
Right. Now I have looked at the problem I have seen this is almost impossible to fix without breaking the public interface of the SearchParameters class.  My conclusion after looking at it: it's broken but useable (the parameters will end up in the wrong collection).  As well, there are some parameters that could have modifiers which cannot be set (except by adding them to the general parameters collection).

As a result, I'll have to create a new class as to not break the current API and deprecate this one. I have started work on it, but won't be able to finish it for this milestone.....
